cmds/rmmod/rmmod.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Remove a module from the Linux kernel
+//
+// Synopsis:
+//	rmmod name
+//
+// Description:
+//	rmmod is a clone of rmmod(8)
+//
+// Author:
+//     Roland Kammerer <dev.rck@gmail.com>
+package main
+
+import (
+	"log"
+	"os"
+	"syscall"
+	"unsafe"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		log.Fatalf("rmmod: ERROR: missing module name.\n")
+	}
+
+	modname := os.Args[1]
+	flags := syscall.O_NONBLOCK
+	ret, _, err := syscall.Syscall(syscall.SYS_DELETE_MODULE, uintptr(unsafe.Pointer(&[]byte(modname)[0])), uintptr(len(modname)), uintptr(unsafe.Pointer(&flags)))
```

Comment:
https://golang.org/pkg/syscall/#BytePtrFromString

Can be used for the go string to c string conversion

Reply:
The man page I was just looking at just takes a string and flags - why is this also taking the length of the module name and passing a pointer to flags?

Reply:
The string to byte pointer was taken from the existing insmod, but we could/should switch that, I agree.

For the parameters you are right, it should be something like:
```
syscall.Syscall(syscall.SYS_DELETE_MODULE, uintptr(unsafe.Pointer(&[]byte(modname)[0])),  uintptr(unsafe.Pointer(&flags)), 0)
```

But I think the flags have to be passed as pointer, otherwise they do not match the the type signature.